### PR TITLE
Color utils docs

### PR DESCRIPTION
**Pull request by mdo**
- Splits background utils from color utils for our docs with a new page.
- Adds some Sass docs for variables, maps, and mixins

The `scss-docs` shortcode is amazing, but one limitation is it doesn't seem to work with indentation at the beginning of the single line comments to identify sections. This is what I ran into in #32747 with documenting the utilities map. Curious if @MartijnCuppens or @XhmikosR has some ideas for how to address that.

Also documents the existence of `.bg-body`, closing the docs part of #32810.

Still a little copyediting to refine here, so consider this one a little WIP still.

- [x] Document CSS custom properties somewhere (under Sass? another h2 called CSS?)
- [x] Other vars, mixins, etc to add?
- [x] Revert the indentation changes and properly indent capture start/end statements

### CODE REVIEW

**Comment by mdo** [comment]
@XhmikosR Elaborating per Slack... the latest commit here should break the build because there's indentation before the comment lines.

File: scss/_utilities.scss [hunk 1]
```diff
@@ -24,13 +24,15 @@ $utilities: map-merge(
       property: overflow,
       values: auto hidden visible scroll,
     ),
+    // scss-docs-start display-utilities
```
Comment (XhmikosR) [inline scss/_utilities.scss:27]: This doesn't seem to be called right now, so I assumed a few things.

**Comment by XhmikosR** [comment]
@mdo see if it works and we'll clean up the branch later. Also check the OP TODO.

I'll look into improving the shortcode later in #32781

**Comment by mdo** [comment]
Looking solid @XhmikosR, thanks!

One thing for your referenced PR, it'd be awesome to un-indent these lines if possible. Right now the snippet brings in those extra spaces. That might be too difficult, so not a huge deal if so.

**Comment by XhmikosR** [comment]
>One thing for your referenced PR, it'd be awesome to un-indent these lines if possible. Right now the snippet brings in those extra spaces. That might be too difficult, so not a huge deal if so.

Yup, I know, it's in my TODO after this one lands :)

File: site/data/sidebar.yml [hunk 2]
```diff
@@ -92,6 +92,7 @@
 - title: Utilities
   pages:
     - title: API
+    - title: Background
     - title: Borders
     - title: Colors
```
Comment (ffoodd) [inline site/data/sidebar.yml:97]: Shouldn't this be renamed as `Color` for consistency? Questionning only, no opinion.

**Review by ffoodd — APPROVED** [review]

File: site/data/sidebar.yml (hunk 2 above)
Comment (XhmikosR) [reply site/data/sidebar.yml:97]: We actually seem to mix and match these... we have Interactions, shadows etc. We should probably streamline these at some point and add redirects.

**Pull request merged**